applications/nrf5340_audio/src/audio/streamctrl_unicast_client.c

```diff
@@ -0,0 +1,577 @@
+		case BUTTON_PLAY_PAUSE:
+
+			if (strm_state == STATE_STREAMING) {
+				ret = bt_content_ctrl_stop(NULL);
+				if (ret) {
+					LOG_WRN("Could not stop: %d", ret);
+				}
+
+			} else if (strm_state == STATE_PAUSED) {
+				ret = bt_content_ctrl_start(NULL);
+				if (ret) {
+					LOG_WRN("Could not start: %d", ret);
+				}
+
+			} else {
+				LOG_WRN("In invalid state: %d", strm_state);
+			}
+
+			break;
+
+		case BUTTON_VOLUME_UP:
+			ret = bt_rend_volume_up();
+			if (ret) {
+				LOG_WRN("Failed to increase volume: %d", ret);
+			}
+
+			break;
+
+		case BUTTON_VOLUME_DOWN:
+			ret = bt_rend_volume_down();
+			if (ret) {
+				LOG_WRN("Failed to decrease volume: %d", ret);
+			}
+
+			break;
+
+		default:
+			LOG_WRN("Unexpected/unhandled button id: %d", msg.button_pin);
+		}
+
+		STACK_USAGE_PRINT("button_msg_thread", &button_msg_sub_thread_data);
+	}
+}
+
+/**
+ * @brief	Handle Bluetooth LE audio events.
+ */
+static void le_audio_msg_sub_thread(void)
+{
+	int ret;
+	const struct zbus_channel *chan;
+
+	while (1) {
+		ret = zbus_sub_wait(&le_audio_evt_sub, &chan, K_FOREVER);
+		ERR_CHK(ret);
+
+		struct le_audio_msg msg;
+
+		ret = zbus_chan_read(chan, &msg, ZBUS_READ_TIMEOUT_MS);
+		ERR_CHK(ret);
+
+		uint8_t event = msg.event;
+
+		LOG_DBG("Received event = %d, current state = %d", event, strm_state);
+
+		switch (event) {
+		case LE_AUDIO_EVT_STREAMING:
+			LOG_DBG("LE audio evt streaming");
+
+			if (strm_state == STATE_STREAMING) {
+				LOG_DBG("Got streaming event in streaming state");
+				break;
+			}
+
+			audio_system_start();
+			stream_state_set(STATE_STREAMING);
+
+			ret = led_blink(LED_APP_1_BLUE);
+			ERR_CHK(ret);
+			break;
+
+		case LE_AUDIO_EVT_NOT_STREAMING:
+			LOG_DBG("LE audio evt not_streaming");
+
+			if (strm_state == STATE_PAUSED) {
+				LOG_DBG("Got not_streaming event in paused state");
+				break;
+			}
+
+			stream_state_set(STATE_PAUSED);
+			audio_system_stop();
+
+			ret = led_on(LED_APP_1_BLUE);
+			ERR_CHK(ret);
+			break;
+
+		default:
+			LOG_WRN("Unexpected/unhandled le_audio event: %d", event);
+			break;
+		}
+
+		STACK_USAGE_PRINT("le_audio_msg_thread", &le_audio_msg_sub_thread_data);
+	}
+}
+
+/**
+ * @brief	Zbus listener to receive events from bt_mgmt.
+ *
+ * @param[in]	chan	Zbus channel.
+ *
+ * @note	Will in most cases be called from BT_RX context,
+ *		so there should not be too much processing done here.
+ */
+static void bt_mgmt_evt_handler(const struct zbus_channel *chan)
+{
+	int ret;
+	const struct bt_mgmt_msg *msg;
+
+	msg = zbus_chan_const_msg(chan);
+	uint8_t event = msg->event;
+
+	switch (event) {
+	case BT_MGMT_CONNECTED:
+		LOG_INF("Device connected");
+		break;
+
+	case BT_MGMT_SECURITY_CHANGED:
+		LOG_INF("Security changed");
+
+		ret = bt_rend_discover(msg->conn);
+		if (ret) {
+			LOG_WRN("Failed to discover rendering services");
+		}
+
+		if (IS_ENABLED(CONFIG_STREAM_BIDIRECTIONAL)) {
+			ret = unicast_client_discover(msg->conn, UNICAST_SERVER_BIDIR);
```

Comment:
We have the technology, but since this streamctrl/app is for a unicast_client similar to the one we used to have we don't have the option here. That would be its own application